File: 8th/04.26.19_Контрольная_работа/pr4.ml
open String;;
type t = B of t list;;

let rec to_str t n=  
let rec f l k = 
match l with
 [] -> "\n"
|(B li)::tl -> (make (k*n) ' ')^"B [\n"^(f li (k+1))^(make (k*n+2) ' ')^"];\n"^(f tl k) in
f [t] 0;;

print_string (to_str (B[B[B[];B[]];B[];B[]]) (read_int()));; 

File: 9th/PYHTHON/20.05.2020 HW/pr4.py
import re

f = open("snames","r")

l = []

for x in f:
	a = re.match("^([A-Z][a-z]+)\n$",x)
	b = re.match("^([A-Z][a-z]+) ([A-Z][a-z]+)$\n",x)
	c = re.match("^([A-Z][a-z]+) ([A-Z][a-z]+) ([A-Z][a-z]+)$\n",x)
	if a:
		l+=[("-",a.group(1),"-")]
	elif b:
		l+=[(b.group(1),b.group(2),"-")]
	elif c:
		l+=[(c.group(1),c.group(2),c.group(3))]

print (l)
	